experiments/experiments.go

```diff
@@ -0,0 +1,195 @@
+package experiments
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"hash"
+	"io"
+	"math/big"
+	"time"
+
+	"github.com/reddit/baseplate.go/filewatcher"
+	"github.com/reddit/baseplate.go/log"
+)
+
+type Document map[string]*ExperimentConfig
+
+type Experiments struct {
+	watcher *filewatcher.Result
+}
+
+func NewExperiments(ctx context.Context, path string, logger log.Wrapper) (*Experiments, error) {
+	parser := func(r io.Reader) (interface{}, error) {
+		var doc Document
+		err := json.NewDecoder(r).Decode(&doc)
+		if err != nil {
+			return nil, err
+		}
+		return doc, nil
+	}
+	result, err := filewatcher.New(ctx, path, parser, logger)
+	if err != nil {
+		return nil, err
+	}
+	return &Experiments{
+		watcher: result,
+	}, nil
+}
+
+type Bucket struct {
+}
+
+type Experiment interface {
+	UniqueID(map[string]string) string
+	Variant(string) (string, error)
+	LogBucketing() bool
+}
+
+func (e *Experiments) Variant(name string, bucketKey string, bucketingEventOverride bool) (string, error) {
+	experiment, err := e.experiment(name)
+	if err != nil {
+		return "", err
+	}
+	return experiment.Variant(bucketKey)
+}
+
+func (e *Experiments) experiment(name string) (Experiment, error) {
+	result := e.watcher.Get()
+	doc, ok := result.(Document)
+	if !ok {
+		return nil, fmt.Errorf("unexpected type %T", doc)
+	}
+	experiment, ok := doc[name]
+	if !ok {
+		return nil, ErrorUnknownExperiment(name)
+	}
+	if isSimpleExperiment(experiment.Type) {
+		return NewSimpleExperiment(experiment), nil
+	}
+	return nil, nil
+}
+
+type ParsedExperiment struct {
+	ExperimentVersion int       `json:"experiment_version"`
+	ShuffleVersion    string    `json:"shuffle_version"`
+	BucketVal         string    `json:"bucket_val"`
+	LogBucketing      bool      `json:"log_bucketing"`
+	Variants          []Variant `json:"variants"`
+	BucketSeed        string    `json:"bucket_seed"`
+}
+
+type ExperimentConfig struct {
+	// ID is the experiment identifier and should be unique for each experiment.
+	ID int `json:"id`
+	// Name is the experiment name and should be unique for each experiment.
+	Name           string           `json:"name"`
+	Owner          string           `json:"owner"`
+	Enabled        *bool            `json:"enabled"`
+	Version        string           `json:"version"`
+	Type           string           `json:"type"`
+	StartTimestamp int64            `json:"start_ts"`
+	StopTimestamp  int64            `json:"stop_ts"`
```

Comment:
We should define a custom type in `timebp`, similar to `TimestampMillisecond` there.

Reply:
Can we reuse `timebp.TimestampMillisecond` here?

Reply:
but here it's seconds, not milliseconds?